src/main/java/com/example/bookshop/controller/BookingController.java

```diff
@@ -0,0 +1,114 @@
+package com.example.bookshop.controller;
+
+import com.example.bookshop.entity.Booking;
+import com.example.bookshop.entity.Product;
+import com.example.bookshop.entity.Status;
+import com.example.bookshop.pojos.BookingObj;
+import com.example.bookshop.pojos.ProductObj;
+import com.example.bookshop.pojos.StatusObj;
+import com.example.bookshop.pojos.UserObj;
+import com.example.bookshop.service.IBookingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+@RequestMapping("/booking")
+public class BookingController {
+    @Autowired
+    private IBookingService bookingService;
+
+
+    /**
+     * Create new booking in DB
+     * @param status of booking
+     * @param description - String
+     * @return a JSON with created booking
+     */
+    @PostMapping("/create")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Booking createBooking(@RequestParam StatusObj status,
+                                 @RequestParam String description,
+                                 @RequestParam ProductObj product,
+                                 @RequestParam UserObj user){
+        return bookingService.createBooking(new BookingObj(status, description, product, user));
+    }
+
+
+    /**
+     * Create one new booking in DB
+     * @param bookingObj Object
+     * @return a JSON with created booking
+     */
+    @PostMapping("/createBooking")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Booking createBooking(@RequestBody BookingObj bookingObj) {
```

Comment:
seems strange to me that you have BookingObj (better use DTO naming) as input and Booking as output
If you have DTOs, better to communicate with external world using them

Imagine situation. You are frontend developer. You have CreateBooking page. User clicks 'Book button'. As a developer you send BookingDTO to backend, but receive Booking entity in response. On next page your user realizes that he needs to update his booking. ANd as a developer you have to transform your Booking entity to BookingDto (cause your EDIT api takes BookingDto as input)

Reply:
agree with you. Thanks